src/containers/tud/TimeUseDiarySagas.js

```diff
@@ -213,17 +220,67 @@ function* getSubmissionsByDateWatcher() :Saga<*> {
   yield takeEvery(GET_SUBMISSIONS_BY_DATE, getSubmissionsByDateWorker);
 }
 
-function* downloadTudResponsesWorker(action :SequenceAction) :Saga<WorkerResponse> {
+function* downloadSummarizedData(entities, outputFilename) :Saga<WorkerResponse> {
+  let workerResponse = {};
+  try {
+
+    const submissionIds = entities.map((entity) => entity.get(OPENLATTICE_ID_FQN)).flatten();
+    const submissionMetadata = Map(entities.map((entity) => [entity.getIn([OPENLATTICE_ID_FQN, 0]), entity]));
+
+    // entity set ids
+    const surveyEntitySets :Map = yield select(selectEntitySetsByModule(AppModules.QUESTIONNAIRES));
+
+    const submissionESID = surveyEntitySets.get(SUBMISSION);
+    const registeredForESID = surveyEntitySets.get(REGISTERED_FOR);
+    const summarysetESID = surveyEntitySets.get(SUMMARY_SET);
+
+    // filtered search on submissions
+    const filter = {
+      destinationEntitySetIds: [submissionESID],
+      edgeEntitySetIds: [registeredForESID],
+      entityKeyIds: submissionIds.toJS(),
+      sourceEntitySetIds: [summarysetESID]
+    };
+
+    const response = yield call(
+      searchEntityNeighborsWithFilterWorker,
+      searchEntityNeighborsWithFilter({
+        entitySetId: submissionESID,
+        filter,
+      })
+    );
+    if (response.error) throw response.error;
+
+    const csvHeaders = OrderedSet().asMutable();
```

Comment:
No change required, just a thought. I'm not sure if the search response will always have the same order of neighbors. It seems possible that you'll get a different order of headers over time, which could potentially break any scripts researchers create that depend on the order of the columns.